### PR TITLE
feat: Identity components refactor

### PR DESCRIPTION
**Pull request by kyhyco**
**What changed? Why?**
- Users can now decide to render attestations via passing `Badge` component to either `Avatar` or `Name` components.
- Add `Address` component and remove `showAddress` logic from `Name` component
- Update default avatar
- Replace `WithAvatarBadge` and `WithNameBadge` with `DisplayBadge`

**Notes to reviewers**
TODO: update docs after new release. (includes root doc index.mdx)

**How has it been tested?**
locally

### CODE REVIEW

File: src/identity/components/defaultSVGs.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+export const defaultLoadingSVG = (
```
Comment (Zizzamia) [inline src/identity/components/defaultSVGs.tsx:1]: nit, split in two files.

File: site/docs/pages/identity/introduction.mdx [hunk 2]
```diff
@@ -48,7 +49,7 @@ The available components are:
         <b>
           <Name address="0x838aD0EAE54F99F1926dA7C3b6bFbF617389B4D9" />
         </b>
-        <Name address="0x838aD0EAE54F99F1926dA7C3b6bFbF617389B4D9" showAddress />
+        {/* <Address address="0x838aD0EAE54F99F1926dA7C3b6bFbF617389B4D9" /> */}
```
Comment (kyhyco) [inline site/docs/pages/identity/introduction.mdx:52]: uncomment this and import `Address` after new release

File: site/docs/pages/identity/address.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{/*import { Address } from '@coinbase/onchainkit/identity'; */}
```
Comment (kyhyco) [inline site/docs/pages/identity/address.mdx:1]: uncomment and add this below
```
<App>
  <Address address="0x02feeb0AdE57b6adEEdE5A4EEea6Cf8c21BeB6B1" />
</App>
```

File: site/docs/pages/identity/identity.mdx [hunk 4]
```diff
@@ -1,5 +1,5 @@
-{/* import { Avatar, Identity, Name } from '../../../../src/identity'; */}
-import { Avatar, Identity, Name } from '@coinbase/onchainkit/identity';
+{/* import { Avatar, Identity, Name, Badge } from '../../../../src/identity'; */}
+import { Avatar, Identity, Name, Badge } from '@coinbase/onchainkit/identity';
```
Comment (kyhyco) [inline site/docs/pages/identity/identity.mdx:2]: import `Address` and add `<Address />` below

File: site/docs/pages/identity/introduction.mdx [hunk 5]
```diff
@@ -16,6 +16,7 @@ OnchainKit provides TypeScript utilities and React components to help you build
 
 The available components are:
 
+- [`<Address />`](/identity/address): Display an account address.
```
Comment (Zizzamia) [inline site/docs/pages/identity/introduction.mdx:19]: I am thinking more and more to kill Intro pages and just conver them into Guides.

Nothing to do now. But more something I am thinking more and more.

cc @kyhyco @fakepixels @abcrane123

File: src/identity/components/Address.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  const { address: contextAddress } = useIdentityContext();
+  if (!contextAddress && !address) {
+    throw new Error(
+      'Address: an Ethereum address must be provided to the Identity or Address component.',
+    );
+  }
+
+  return (
+    <span className={cn(text.label2, className)}>
+      {getSlicedAddress(contextAddress ?? address)}
```
Comment (Zizzamia) [inline src/identity/components/Address.tsx:16]: nit. for later, we might need an hook for this, so it does not cause a new instance of this function at every render.

File: src/identity/components/Avatar.tsx [hunk 7]
```diff
@@ -33,83 +37,53 @@ export function Avatar({
...
-              from="0 50 50"
-              to="360 50 50"
-              dur="1s"
-              repeatCount="indefinite"
-            />
-          </circle>
-        </svg>
-      )
-    );
+    return loadingComponent || defaultLoadingSVG;
```
Comment (Zizzamia) [inline src/identity/components/Avatar.tsx:51]: fancy

File: src/identity/components/Badge.tsx [hunk 8]
```diff
@@ -11,7 +11,7 @@ export function Badge({
   tickerColor = 'white',
```
Comment (Zizzamia) [inline src/identity/components/Badge.tsx:11]: nit for later, we need to kill those props. in favour of using classNames.

File: src/identity/components/DisplayBadge.test.tsx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  useOnchainKit: jest.fn(),
+}));
+jest.mock('../hooks/useAttestations', () => ({
+  useAttestations: jest.fn(),
+}));
+jest.mock('../context', () => ({
+  useIdentityContext: jest.fn(),
+}));
+
+describe('DisplayBadge', () => {
```
Comment (Zizzamia) [inline src/identity/components/DisplayBadge.test.tsx:23]: love the tests

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**